internal/charmstore/addentity.go

```diff
@@ -543,10 +543,6 @@ func (s *Store) addEntity(entity *mongodoc.Entity) (err error) {
 			}
 		}
 	}()
```

Comment:
We no longer need this deferred func now, as nothing happens after this point anymore.